src/main/java/frc/robot/commands/DriveToPosition.java

```diff
@@ -0,0 +1,94 @@
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.SwerveDrive;
+
+public class DriveToPosition extends CommandBase {
+  private final double k_maxSpeedMps = 4;
+  private final double k_slowDownDistanceM = 1;
+  private final double k_maxRotationChange = 1;
+  private final double k_slowDownAngleDegrees = 10;
+  SwerveDrive m_drive;
+  double m_x, m_y, m_thetaDegrees;
+  Pose2d m_targetPose;
+  /** Creates a new DriveToPosition. */
+  public DriveToPosition(SwerveDrive drive, double x, double y, double thetaDegrees) {
+    // Use addRequirements() here to declare subsystem dependencies.
+    addRequirements(drive);
+    m_drive = drive;
+    m_x = x;
+    m_y = y;
+    m_thetaDegrees = thetaDegrees;
+    m_targetPose = new Pose2d(x, y, Rotation2d.fromDegrees(thetaDegrees));
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    var difference = getDistanceFromTarget();
+    SmartDashboard.putString("difference", difference.toString());
+    double diffX = difference.getX();
+    double diffY = difference.getY();
+    double distance = Math.sqrt(Math.pow(diffX, 2) + Math.pow(diffY, 2));
+    double speedX = k_maxSpeedMps * diffX / distance;
+    double speedY = k_maxSpeedMps * diffY / distance;
+    if(distance < k_slowDownDistanceM) {
+      double slowDownRatio = distance / k_slowDownDistanceM;
+      speedX *= slowDownRatio;
+      speedY *= slowDownRatio;
+    }
+    var rotationDifference = getRotationFromTarget();
+    SmartDashboard.putString("rotationDifference", rotationDifference.toString());
+    var rotationDifferenceDegrees = rotationDifference.getDegrees();
+    var rotationChangeSpeed = rotationDifferenceDegrees < 0 ? -k_maxRotationChange : k_maxRotationChange;
+    if(rotationDifferenceDegrees < k_slowDownAngleDegrees) {
+      double slowDownRatio = rotationDifferenceDegrees / k_slowDownAngleDegrees;
+      rotationChangeSpeed *= slowDownRatio;
+    }
+    m_drive.moveFieldRelative(speedX, speedY, rotationChangeSpeed);
+  }
+
+  private Translation2d getDistanceFromTarget() {
+    var currentPose = m_drive.getPose();
+    var difference = m_targetPose.getTranslation().minus(currentPose.getTranslation());
+    return difference;
+  }
+
+  private Rotation2d getRotationFromTarget() {
+    var currentPose = m_drive.getPose();
+    var currentRotation = currentPose.getRotation();
+    var targetRotation = m_targetPose.getRotation();
+    SmartDashboard.putString("targetRotation", targetRotation.toString());
```

Comment:
Can you remove the debug smartdashboard lines?

Reply:
Yes